nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/PutKafka.java

```diff
@@ -371,11 +371,12 @@ private FlowFile cleanUpFlowFileIfNecessary(FlowFile flowFile, ProcessSession se
     /**
      *
      */
-   private Object determinePartition(SplittableMessageContext messageContext, ProcessContext context, FlowFile flowFile) {
+    private Integer determinePartition(SplittableMessageContext messageContext, ProcessContext context,
+            FlowFile flowFile) {
        String partitionStrategy = context.getProperty(PARTITION_STRATEGY).getValue();
-       String partitionValue = null;
+        Integer partitionValue = null;
        if (partitionStrategy.equalsIgnoreCase(USER_DEFINED_PARTITIONING.getValue())) {
-           partitionValue = context.getProperty(PARTITION).evaluateAttributeExpressions(flowFile).getValue();
+            partitionValue = Integer.parseInt(context.getProperty(PARTITION).evaluateAttributeExpressions(flowFile).getValue());
```

Comment:
We should probably wrap the Integer.parseInt() in a try/catch and if an Exception is thrown return null?